content/articles/ssh-tunneling/index.md

```diff
@@ -0,0 +1,130 @@
+- [Remote port forwarding](#remote-port-forwarding)
+- [Conclusion](#conclusion)
+
+### Prerequisites
+To perform SSH tunneling, you need the following:
+- A publicly accessible SSH server.
+- The resources to be accessed.
+- A local device.
+- A remote device for remote port forwarding.
+
+### Local port forwarding
+Local port forwarding allows us access to remote content or resources which we don't have access to. For example, let's say we have a remote Remote Desktop Protocol (RDP) server or a remote database has a firewall that blocks some ports but allows access to some of the ports.
+
+To access the internal network to tunnel through the access ports and smuggle content that we want. Local port forwarding creates a local server and listens on our local machines for a local port and accesses that port that acts as a proxy. 
+
+The image below is a demonstration of local port forwarding:
+
+![Local port forwarding](/engineering-education/ssh-tunneling/local.jpg)
+
+Using the image above, let us have an explanation of SSH tunneling from it. Suppose we have the following devices with IP addresses as shown respectively:
+- An internal resource that we need access to on port 8080 and IP address `192.168.1.3`.
+- A public SSH server of IP address `192.168.1.2` and a public IP address of `44.11.22.33` on port 22.
+- A local device of IP address `10.0.0.4`.
+- Local port 8888.
+
+> Our SSH server and the internal resources are on the same network. This implies that our public SSH server has access to our internal resources.
+
+Our local machine on IP address `10.0.0.4` is on a completely different network and cannot access resources on the IP address `192.168.1.3`. What we need to do is create an SSH tunnel between our local machine and our public SSH server. This is because the IP Address `44.11.22.33` is a public IP address and port 22 is a public open SSH port.
+
+Local port forwarding syntax:
+
+```bash
+ssh -L <local port>:<remote ip address>:<remote port> <public ssh server ip address>
+```
+
+Let us explain the above syntax:
+- -L &rarr; - It is the local port forwarding.
+- 8888 &rarr; - This is the local port.
+- 192.168.1.3 &rarr; - This is a remote IP address.
+- 8080 &rarr; - It is a remote port.
+- 44.11.22.33 &rarr; - It is the public SSH server ip address.
+
+The command below will be used to perform local port forwarding:
+
+```bash
+ssh -L 8888:192.168.1.3:8080 44.11.22.33
+```
+
+We are going to access our public SSH server using SSH. To do this, since our local machine is dumb and is not enough to access the remote resources, we will create a local port, that is, port 8888. 
+
+What happens is that the SSH server will start listening on port 8888 on our local machine. Any traffic that goes to port 8888 from our local machine is forwarded to the IP address `192.168.1.3` on port 8080 through the public SSH server. As a result, a TCP connection is established with our SSH server to our machine because it is open.
```

Comment:
```suggestion
What happens is that the SSH server will listen on port 8888 from our local machine. Any traffic that goes to port 8888 from our local machine is forwarded to the IP address `192.168.1.3` on port 8080 through the public SSH server. As a result, a TCP connection is established with our SSH server to our machine because it is open.